Added autosend data decoding function

Paxy opened the pull request:
Modified sireader.py class SIReaderControl to add autosend_punch() function.
Added example for decoding autosend data and trasfer over UART1 to Meshtastic network.

Paxy committed:
Update sireader2.py

Added autosend_punch function under SIReaderControl class to use existing functions to decode data received by autosend programmed control station.

Paxy committed:
Add files via upload

This example reads the station programmed to autosend the data and redirects to UART1 port to be transfered over Meshtastic network
Sample of read data:
53|233177|12:14:09

per-magnusson commented on sireader2.py:
The indentation of the else clause seems incorrect.

Paxy replied:
function autosend_punch() is just a copy of the function poll_punch() without asking for stored backup.
else clause is also the copied. 
I think i could be omited, but kept it code to remain the same as poll_punch() function.

per-magnusson replied:
Looking at the code, I think there are indentation errors also in poll_punch(). The else clauses in both functions will never be executed (they would only be executed if the while condition became false, which it impossible since the condition is literally "True" and the loops are exited by break (which do not trigger the else clause). 

The purpose of the else clauses must be to raise an exception if anything other than a C_TRANS_REC (extended protocol autosend) packet is received.

(With a slight reservation that I am not the worlds most experienced Python programmer, so I could be missing something, but I do not think I am in this case.)

Paxy replied:
I agree. You can probably safely remove both else clauses.

per-magnusson replied:
I have kept them, but indented them to go with the if-statements. I am pretty sure that is the best option, since the code then raises an exception if something other than an autosend packet shows up.

Paxy commented:
function autosend_punch() is just a copy of the function poll_punch() without asking for stored backup.
else clause is also the copied. 
I think i could be omited, but kept it code to remain the same as poll_punch() function.

Paxy committed:
Update sireader2.py

Removed both else clauses

Paxy reviewed (commented):
Else clauses removed

per-magnusson committed:
Update sireader2.py

Fixed indentation in poll_punch
Added autosend_punch contributed by Paxy (essentially the same as poll_punch, but with station code also included in the return value)
